nemo/collections/asr/models/classification_models.py

```diff
@@ -74,6 +75,29 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
         featurizer = WaveformFeaturizer(
             sample_rate=config['sample_rate'], int_values=config.get('int_values', False), augmentor=augmentor
         )
+
+        if 'vad_stream' in config and config['vad_stream']:
+            print("Perform streaming frame-level VAD")
+            dataset = AudioToSpeechLabelDataSet(
+                manifest_filepath=config['manifest_filepath'],
+                labels=config['labels'],
+                featurizer=featurizer,
+                max_duration=config.get('max_duration', None),
+                min_duration=config.get('min_duration', None),
+                trim=config.get('trim_silence', True),
+                load_audio=config.get('load_audio', True),
+                time_length=config.get('time_length', 0.31),
+                shift_length=config.get('shift_length', 0.01),
+            )
+            return torch.utils.data.DataLoader(
+                dataset=dataset,
+                batch_size=1,
+                collate_fn=dataset.vad_frame_seq_collate_fn,
+                drop_last=config.get('drop_last', False),
+                shuffle=config['shuffle'],
+                num_workers=config.get('num_workers', 0),
+            )
+
```

Comment:
I think you can just use the single return with updated batch_size and collate_func

Reply:
what do you mean single return?
We have two collate_fn in class AudioToSpeechLabelDataSet. If make single return with dataset.collate_fn we need to make a _collate_fn for class `AudioToSpeechLabelDataSet` and decide whether to return `fixed_seq_collate_fn`,  `sliced_seq_collate_fn` or `vad_frame_seq_collate_fn`. Is that what you are suggesting?

Reply:
if vad_stream:
 dataset =AudioToSpeechLabelDataset(args)
bs =1
collate_func=dataset.vad_frame_seq_collate_fn
else:
dataset= AudioLabelDataset(args)
bs=config['batch_size']
collate_func=dataset.collate_fn

return DataLoader(
dataset=dataset,
batch_size=bs,
collate_fn=collate_func,
etc)

Reply:
haha that's what you mean.. Yeah good idea! 👍